File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/SBPacketChatMessage.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;
import pize.tests.voxelgame.server.net.PlayerGameConnection;

import java.io.IOException;

public class SBPacketChatMessage extends IPacket<PlayerGameConnection>{
    
    public static final int PACKET_ID = 19;
    
    public SBPacketChatMessage(){
        super(PACKET_ID);
    }
    
    
    public String message;
    
    public SBPacketChatMessage(String message){
        this();
        this.message = message;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeUTF(message);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        message = stream.readUTF();
    }
    
    @Override
    public void handle(PlayerGameConnection packetHandler){
        packetHandler.handleChatMessage(this);
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/SBPacketChatMessage.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.tests.voxelgame.server.net.PlayerGameConnection
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class SBPacketChatMessage() : IPacket<PlayerGameConnection?>(PACKET_ID) {
    var message: String? = null

    constructor(message: String?) : this() {
        this.message = message
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeUTF(message)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        message = stream!!.readUTF()
    }

    override fun handle(packetHandler: PlayerGameConnection) {
        packetHandler.handleChatMessage(this)
    }

    companion object {
        const val PACKET_ID = 19
    }
}